BDArmory/Ammo/PooledBullet.cs

```diff
@@ -1461,14 +1562,14 @@ private bool ExplosiveDetonation(Part hitPart, RaycastHit hit, Ray ray, bool pen
                     if (fuzeType == BulletFuzeTypes.Timed || fuzeType == BulletFuzeTypes.Flak)
                     {
                         if (HEType != PooledBulletTypes.Slug)
-                            ExplosionFx.CreateExplosion(hit.point, GetExplosivePower(), explModelPath, explSoundPath, ExplosionSourceType.Bullet, caliber, null, sourceVesselName, null, ray.direction, -1, false, bulletMass, -1, dmgMult, HEType == PooledBulletTypes.Shaped ? "shapedcharge" : "standard");
+                            ExplosionFx.CreateExplosion(hit.point, GetExplosivePower(), explModelPath, explSoundPath, ExplosionSourceType.Bullet, caliber, null, sourceVesselName, null, ray.direction, -1, false, bulletMass, -1, dmgMult, HEType == PooledBulletTypes.Shaped ? "shapedcharge" : "standard", null, HEType == PooledBulletTypes.Shaped ? apBulletMod : 1f);
```

Comment:
@SuicidalInsanity, should the direction also be `default` here? Or rather the direction should probably be dependent on `HEType` in these calls since `default` is going to be nonsensical if the bullet type is a shaped charge.